### PR TITLE
Fix/migrate modal iplb

### PR DESCRIPTION
**Pull request by varun257**
### Description of the Change

replace cui-modal with oui-modal
replace form elements with oui-fields

### Benefits

Brings load balancer modals up to date with oui-kit standards

### Possible Drawbacks

None
### Applicable Issues

SCDC-822

### CODE REVIEW

File: client/app/iplb/home/updateQuota/iplb-update-quota.controller.js [hunk 1]
```diff
@@ -12,19 +13,12 @@ class IpLoadBalancerUpdateQuotaCtrl {
...
-        ).then(result => {
-            this.$uibModalInstance.close(result);
-        }).catch(err => {
-            this.$uibModalInstance.close(err);
-        }).finally(() => {
-            this.saving = false;
+        this.update = this.ControllerHelper.request.getHashLoader({
+            loaderFunction: () => this.IpLoadBalancerHomeService.updateQuota(this.$stateParams.serviceName, this.quota.zone, this.alert * Math.pow(1000, 3))
+                .then(response => this.$uibModalInstance.close(response))
+                .catch(response => this.$uibModalInstance.close(response))
```
Comment (radireddy) [inline client/app/iplb/home/updateQuota/iplb-update-quota.controller.js]: for better readability, rename response to error

File: client/app/iplb/modal/failover-ip/iplb-failover-ip-detail.html [hunk 2]
```diff
@@ -1,10 +1,11 @@
-<cui-modal>
-    <cui-modal-header data-on-dismiss="IpLoadBalancerFailoverIpDetailCtrl.dismiss()"></cui-modal-header>
-    <cui-modal-body data-title=":: 'iplb_modal_failover_ip_title' | translate">
-        <oui-spinner data-ng-if="IpLoadBalancerFailoverIpDetailCtrl.isModalLoading()"></oui-spinner>
-
+<oui-modal
+    data-title="{{ ::'iplb_modal_failover_ip_title' | translate }}"
+    data-on-dismiss="IpLoadBalancerFailoverIpDetailCtrl.dismiss()"
+    data-loading="IpLoadBalancerFailoverIpDetailCtrl.isModalLoading()">
+    <div data-ng-if="!IpLoadBalancerFailoverIpDetailCtrl.isModalLoading()">
```
Comment (radireddy) [inline client/app/iplb/modal/failover-ip/iplb-failover-ip-detail.html]: this is not required as data-loading on oui-modal will take care of it

File: client/app/iplb/modal/failover-ip/iplb-failover-ip-detail.html [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
+        <p class="oui-modal__text" data-translate="iplb_modal_nat_ip_description"></p>
         <div data-ng-if="!IpLoadBalancerFailoverIpDetailCtrl.isModalLoading()">
```
Comment (radireddy) [inline client/app/iplb/modal/failover-ip/iplb-failover-ip-detail.html]: same here, not required

File: client/app/iplb/modal/nat-ip/iplb-nat-ip-detail.html [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
-    <cui-modal-header data-on-dismiss="IpLoadBalancerNatIpDetailCtrl.dismiss()"></cui-modal-header>
-    <cui-modal-body data-title=":: 'iplb_modal_nat_ip_title' | translate">
-        <oui-spinner data-ng-if="IpLoadBalancerNatIpDetailCtrl.isModalLoading()"></oui-spinner>
-        <div data-ng-if="!IpLoadBalancerNatIpDetailCtrl.isModalLoading()">
-            <cui-modal-text data-text="'iplb_modal_nat_ip_description' | translate"></cui-modal-text>
+<oui-modal
+    data-title="{{ ::'iplb_modal_nat_ip_title' | translate }}"
+    data-on-dismiss="IpLoadBalancerNatIpDetailCtrl.dismiss()"
+    data-loading="IpLoadBalancerNatIpDetailCtrl.isModalLoading()">
+    <div data-ng-if="!IpLoadBalancerNatIpDetailCtrl.isModalLoading()">
```
Comment (radireddy) [inline client/app/iplb/modal/nat-ip/iplb-nat-ip-detail.html]: same here, not required

File: client/app/iplb/modal/nat-ip/iplb-nat-ip-detail.html [hunk 5]
```diff
@@ -1,16 +1,15 @@
...
+        <p class="oui-modal__text" data-translate="iplb_modal_nat_ip_description"></p>
 
-            <div data-ng-if="!IpLoadBalancerFailoverIpDetailCtrl.isModalLoading()">
-                <div data-ng-repeat="nat in IpLoadBalancerNatIpDetailCtrl.ips.data track by $index">
-                    <label class="oui-label" for="ip-{{nat.zone}}" data-ng-bind="nat.zone"></label>
-                    <cui-clipboard data-id="ip-{{nat.zone}}" data-text="nat.ip"></cui-clipboard>
-                </div>
+        <div data-ng-if="!!IpLoadBalancerNatIpDetailCtrl.isModalLoading()">
```
Comment (radireddy) [inline client/app/iplb/modal/nat-ip/iplb-nat-ip-detail.html]: same here, not required

File: client/app/iplb/modal/offer/iplb-offer-change.html [hunk 6]
```diff
@@ -1,18 +1,18 @@
...
-            <cui-cancel-button data-text="'common_cancel' | translate" data-on-click="IpLoadBalancerOfferChangeCtrl.cancel()"></cui-cancel-button>
-            <cui-submit-button data-text="'common_edit' | translate"
-                data-on-click="IpLoadBalancerOfferChangeCtrl.confirm()"
-                data-disabled="IplbOfferChangeForm.$invalid || IpLoadBalancerOfferChangeCtrl.isModalLoading()"></cui-submit-button>
-        </cui-modal-footer>
-    </form>
-</cui-modal>
+<form id="IplbOfferChangeForm"
+      name="IplbOfferChangeForm"
+      data-ng-submit="IplbOfferChangeForm.$valid &&IpLoadBalancerOfferChangeCtrl.confirm()"
```
Comment (radireddy) [inline client/app/iplb/modal/offer/iplb-offer-change.html]: give space after &&

File: client/app/iplb/modal/offer/iplb-offer-change.html [hunk 7]
```diff
@@ -1,18 +1,18 @@
...
+      novalidate>
+    <oui-modal
+        data-title="{{ ::'iplb_modal_offer_change_title' | translate }}"
+        data-primary-label="{{ ::'common_edit' | translate }}"
+        data-secondary-action="IpLoadBalancerOfferChangeCtrl.cancel()"
+        data-secondary-label="{{ ::'common_cancel' | translate }}"
+        data-on-dismiss="IpLoadBalancerOfferChangeCtrl.cancel()"
+        data-loading="IpLoadBalancerOfferChangeCtrl.isModalLoading()">
+
+        <div data-ng-if="!IpLoadBalancerOfferChangeCtrl.isModalLoading()">
```
Comment (radireddy) [inline client/app/iplb/modal/offer/iplb-offer-change.html]: same here, not required

File: client/app/iplb/modal/failover-ip/iplb-failover-ip-detail.html [hunk 8]
```diff
@@ -1,7 +1,10 @@
...
-    data-title="{{ ::'iplb_modal_failover_ip_title' | translate }}"
-    data-on-dismiss="IpLoadBalancerFailoverIpDetailCtrl.dismiss()"
-    data-loading="IpLoadBalancerFailoverIpDetailCtrl.isModalLoading()">
-        <p class="oui-modal__text" data-translate="iplb_modal_nat_ip_description"></p>
-        <cui-clipboard data-text="ip" data-ng-repeat="ip in IpLoadBalancerFailoverIpDetailCtrl.ips.data track by $index"></cui-clipboard>
-</oui-modal>
+<cui-modal>
+    <cui-modal-header data-on-dismiss="IpLoadBalancerFailoverIpDetailCtrl.dismiss()"></cui-modal-header>
+    <cui-modal-body data-title=":: 'iplb_modal_failover_ip_title' | translate">
+        <oui-spinner data-ng-if="IpLoadBalancerFailoverIpDetailCtrl.isModalLoading()"></oui-spinner>
```
Comment (FredericEspiau) [inline client/app/iplb/modal/failover-ip/iplb-failover-ip-detail.html]: Not centered ?

File: client/app/iplb/modal/offer/iplb-offer-change.html [hunk 9]
```diff
@@ -1,18 +1,17 @@
...
+      data-ng-submit="IplbOfferChangeForm.$valid && IpLoadBalancerOfferChangeCtrl.confirm()"
+      novalidate>
+    <oui-modal
+        data-title="{{ ::'iplb_modal_offer_change_title' | translate }}"
+        data-primary-label="{{ ::'common_edit' | translate }}"
+        data-secondary-action="IpLoadBalancerOfferChangeCtrl.cancel()"
+        data-secondary-label="{{ ::'common_cancel' | translate }}"
+        data-on-dismiss="IpLoadBalancerOfferChangeCtrl.cancel()"
+        data-loading="IpLoadBalancerOfferChangeCtrl.isModalLoading()">
+        <p class="oui-modal__text">
```
Comment (jleveugle) [inline client/app/iplb/modal/offer/iplb-offer-change.html]: Should be remove, right?

File: client/app/iplb/modal/offer/iplb-offer-change.html [hunk 10]
```diff
@@ -1,18 +1,17 @@
...
+            Old offer => {{ IpLoadBalancerOfferChangeCtrl.configuration.data.offer }}
+        </p>
+        <p class="oui-modal__text" data-text="'EDIT SOME OFFER HERE'"></p>
```
Comment (jleveugle) [inline client/app/iplb/modal/offer/iplb-offer-change.html]: I don't understand this line too

Comment (varun257) [reply client/app/iplb/modal/offer/iplb-offer-change.html]: @jleveugle The offer change part is not implemented yet. So they have put some dummy text which i have retained as it is and changed only the modal part

File: client/app/iplb/modal/offer/iplb-offer-change.html (hunk 9 above)
Comment (varun257) [reply client/app/iplb/modal/offer/iplb-offer-change.html]: The modal is not completely implemented and is not shown. Have just changed cui-modal to oui-modal here.

**Comment by varun257** [comment]
@FredericEspiau @jleveugle  The modal in offer-change is not implemented / not being used. Since cui-modal will be deleted soon. Since i dont know whether someone was working on it, so i retained the code and did not delete it.
iplb-failover-ip-detail will be taken up as a new task since QA is not able to find an account which has failover IP. So they cannot test it.

**Comment by antleblanc** [comment]
@varun257 there is conflict.

**Comment by varun257** [comment]
@antleblanc resolved

File: client/app/iplb/modal/offer/iplb-offer-change.html (hunk 10 above)
Comment (cbourgois) [reply client/app/iplb/modal/offer/iplb-offer-change.html]: IMO if it's not implemented, it should not be here.

Comment (varun257) [reply client/app/iplb/modal/offer/iplb-offer-change.html]: @cbourgois and @jleveugle  have removed the modal and added a todo for its implementation

File: client/app/iplb/serverFarm/probe/iplb-server-farm-probe.html [hunk 11]
```diff
@@ -1,103 +1,77 @@
...
-                    </div>
-                </div>
-            </div>
+        <oui-field data-label="{{ ::'iplb_farm_edit_probe_match' | translate }}" data-ng-if="IpLoadBalancerServerFarmProbeEditCtrl.isFieldVisible('match')">
+            <oui-radio-toggle-group data-model="IpLoadBalancerServerFarmProbeEditCtrl.farmProbe.match">
+                <oui-radio
+                    data-text="{{ ::('iplb_farm_edit_probe_match_' + match) | translate }}"
+                    data-value="match"
+                    data-ng-repeat="match in IpLoadBalancerServerFarmProbeEditCtrl.getMatches()"
+                    data-ng-hide="IpLoadBalancerServerFarmProbeEditCtrl.farmProbe.type !== 'http' && match === 'status'">
```
Comment (FredericEspiau) [inline client/app/iplb/serverFarm/probe/iplb-server-farm-probe.html]: Can the `oui-radio-toggle-group` be empty ? Wouldn't it be an issue ? Also, can you use `ngIf` instead? Hiding controls to the user but still having them is worrisome to me

File: client/app/iplb/serverFarm/probe/iplb-server-farm-probe.html [hunk 12]
```diff
@@ -1,103 +1,77 @@
...
-                        <input type="text" id="probe_pattern" name="probe_pattern" class="oui-input" data-ng-model="IpLoadBalancerServerFarmProbeEditCtrl.farmProbe.pattern" />
-                    </div>
-                </div>
-            </div>
+        <oui-field data-label="{{ ::'iplb_farm_edit_probe_match' | translate }}" data-ng-if="IpLoadBalancerServerFarmProbeEditCtrl.isFieldVisible('match')">
+            <oui-radio-toggle-group data-model="IpLoadBalancerServerFarmProbeEditCtrl.farmProbe.match">
+                <oui-radio
+                    data-text="{{ ::('iplb_farm_edit_probe_match_' + match) | translate }}"
+                    data-value="match"
+                    data-ng-repeat="match in IpLoadBalancerServerFarmProbeEditCtrl.getMatches()"
```
Comment (FredericEspiau) [inline client/app/iplb/serverFarm/probe/iplb-server-farm-probe.html]: `track by`

File: client/app/iplb/home/updateQuota/iplb-update-quota.controller.js [hunk 13]
```diff
@@ -12,19 +13,12 @@ class IpLoadBalancerUpdateQuotaCtrl {
...
-            // Convert GiB to bytes
-            this.alert * Math.pow(1000, 3)
-        ).then(result => {
-            this.$uibModalInstance.close(result);
-        }).catch(err => {
-            this.$uibModalInstance.close(err);
-        }).finally(() => {
-            this.saving = false;
+        this.update = this.ControllerHelper.request.getHashLoader({
+            loaderFunction: () => this.IpLoadBalancerHomeService.updateQuota(this.$stateParams.serviceName, this.quota.zone, this.alert * Math.pow(1000, 3))
```
Comment (FredericEspiau) [inline client/app/iplb/home/updateQuota/iplb-update-quota.controller.js]: Maybe you could have `1000 ** 3` 😃

File: client/app/iplb/frontends/preview/iplb-frontends-preview.html [hunk 14]
```diff
@@ -1,75 +1,75 @@
...
+        <dt class="preview-list__term"
+            data-translate="iplb_frontend_preview_datacenter"></dt>
+        <dd class="preview-list__definition"
+            data-ng-bind="IpLoadBalancerFrontendPreviewCtrl.frontend.region.microRegion.text"></dd>
+        <dt class="preview-list__term"
+            data-translate="iplb_frontend_preview_default_farm"></dt>
+        <dd class="preview-list__definition"
+            data-ng-bind="IpLoadBalancerFrontendPreviewCtrl.frontend.defaultFarmId"></dd>
+        <dt class="preview-list__term"
+            data-ng-show="IpLoadBalancerFrontendPreviewCtrl.frontend.ssl"
```
Comment (marie-j) [inline client/app/iplb/frontends/preview/iplb-frontends-preview.html]: Prefer using `data-ng-if`

File: client/app/iplb/frontends/preview/iplb-frontends-preview.html [hunk 15]
```diff
@@ -1,75 +1,75 @@
...
+            data-translate="iplb_frontend_preview_default_certificate"></dt>
+        <dd class="preview-list__definition"
+            data-ng-show="IpLoadBalancerFrontendPreviewCtrl.frontend.ssl"
```
Comment (marie-j) [inline client/app/iplb/frontends/preview/iplb-frontends-preview.html]: `data-ng-if`

File: client/app/iplb/frontends/preview/iplb-frontends-preview.html [hunk 16]
```diff
@@ -1,75 +1,75 @@
...
-</cui-modal>
+<oui-modal
+    data-title="{{ ::'iplb_frontend_preview_title' | translate }}"
+    data-on-dismiss="IpLoadBalancerFrontendPreviewCtrl.dismiss()"
+>
+    <dl class="preview-list preview-list_no-margin">
+        <dt class="preview-list__term"
+            data-translate="iplb_frontend_preview_name"></dt>
+        <dd class="preview-list__definition"
+            data-ng-bind="IpLoadBalancerFrontendPreviewCtrl.frontend.displayName"></dd>
```
Comment (marie-j) [inline client/app/iplb/frontends/preview/iplb-frontends-preview.html]: `::`

File: client/app/iplb/frontends/preview/iplb-frontends-preview.html [hunk 17]
```diff
@@ -1,75 +1,75 @@
...
+        <dt class="preview-list__term"
+            data-translate="iplb_frontend_preview_active"></dt>
+        <dd class="preview-list__definition"
+            data-ng-bind="IpLoadBalancerFrontendPreviewCtrl.frontend.disabled ? 'false' : 'true'"></dd>
+        <dt class="preview-list__term"
+            data-translate="iplb_frontend_preview_id"></dt>
+        <dd class="preview-list__definition"
+            data-ng-bind="IpLoadBalancerFrontendPreviewCtrl.frontend.frontendId"></dd>
+    </dl>
+</oui-modal>
```
Comment (marie-j) [inline client/app/iplb/frontends/preview/iplb-frontends-preview.html]: Could you please replace the `data-ng-show` by `data-ng-if` and add `::` on `data-ng-bind` wherever it's possible

File: client/app/iplb/task/preview/iplb-task-preview.html [hunk 18]
```diff
@@ -1,27 +1,26 @@
...
-                data-ng-bind="$ctrl.task.id"></dd>
-        </dl>
-    </cui-modal-body>
-</cui-modal>
+<oui-modal
+    data-title="{{ ::'iplb_task_preview_title' | translate }}"
+    data-on-dismiss="$ctrl.dismiss()">
+    <dl class="preview-list preview-list_no_margin">
+        <dt class="preview-list__term"
+            data-ng-bind="'iplb_task_preview_creation_date' | translate"></dt>
```
Comment (marie-j) [inline client/app/iplb/task/preview/iplb-task-preview.html]: `data-ng-bind` -> `data-translate`

**Comment by varun257** [comment]
@cbourgois @marie-j @FredericEspiau have fixed all issues. Please have a look

File: client/app/iplb/home/updateQuota/iplb-update-quota.controller.js [hunk 19]
```diff
@@ -7,14 +7,14 @@ class IpLoadBalancerUpdateQuotaCtrl {
         this.quota = Object.assign({}, quota);
 
         // Convert bytes to GiB
-        this.alert = this.quota.alert / Math.pow(1000, 3);
+        this.alert = this.quota.alert / (1000 ** 3);
```
Comment (FredericEspiau) [inline client/app/iplb/home/updateQuota/iplb-update-quota.controller.js]: Does it actually works ? I know I suggested it but I'm not sure our configuration of BabelJS transpiles it correctly 😃

Comment (FredericEspiau) [reply client/app/iplb/home/updateQuota/iplb-update-quota.controller.js]: Nice to know !

**Pull request merged**